client/src/main/scala/drt/client/SPAMain.scala

```diff
@@ -1,37 +1,59 @@
 package drt.client
 
 import java.util.UUID
-
 import diode.Action
 import drt.client.actions.Actions._
-import drt.client.components.{AlertsPage, EditKeyCloakUser, EditKeyCloakUserPage, GlobalStyles, KeyCloakUsersPage, Layout, StatusPage, TerminalComponent, TerminalPlanningComponent, TerminalsDashboardPage}
+import drt.client.components.TerminalDesksAndQueues.{ViewDeps, ViewRecs, ViewType}
+import drt.client.components.{AlertsPage, EditKeyCloakUserPage, GlobalStyles, KeyCloakUsersPage, Layout, StatusPage, TerminalComponent, TerminalPlanningComponent, TerminalsDashboardPage}
 import drt.client.logger._
-import drt.client.modules.GoogleEventTracker
 import drt.client.services.JSDateConversions.SDate
 import drt.client.services._
 import drt.client.services.handlers.GetFeedStatuses
-import drt.shared.CrunchApi.MillisSinceEpoch
 import drt.shared.SDateLike
 import japgolly.scalajs.react.Callback
 import japgolly.scalajs.react.extra.router._
 import org.scalajs.dom
-
+import scalacss.ProdDefaults._
 import scala.collection.immutable.Seq
 import scala.scalajs.js.annotation.{JSExport, JSExportTopLevel}
-import scalacss.ProdDefaults._
 
 object SPAMain {
 
   sealed trait Loc
 
+  sealed trait UrlParameter {
+    val name: String
+    val value: Option[String]
+  }
+
+  case class UrlDateParameter(value: Option[String]) extends UrlParameter {
+    val name = "date"
+  }
+
+  case class UrlTimeRangeStart(value: Option[String]) extends UrlParameter {
+    val name = "timeRangeStart"
+  }
+
+  case class UrlTimeRangeEnd(value: Option[String]) extends UrlParameter {
+    val name = "timeRangeEnd"
+  }
+
+  case class UrlViewType(viewType: Option[ViewType]) extends UrlParameter {
+    val name = "viewType"
+    override val value: Option[String] = viewType.map(_.queryParamsValue)
+  }
+
   case class TerminalPageTabLoc(
-                                 terminal: String,
-                                 mode: String = "current",
-                                 subMode: String = "desksAndQueues",
-                                 date: Option[String] = None,
-                                 timeRangeStartString: Option[String] = None,
-                                 timeRangeEndString: Option[String] = None
-                               ) extends Loc {
+                                  terminal: String,
+                                  mode: String = "current",
+                                  subMode: String = "desksAndQueues",
+                                  queryParams: Map[String, String] = Map.empty[String, String]
+                                ) extends Loc {
+    val date: Option[String] = queryParams.get(UrlDateParameter(None).name).filter(_.matches(".+"))
```

Comment:
I think since we have to instantiate these case classes just to get at their name values, it'd probably be a good idea to put all the different 'name' values in an object and have the case classes refer to those. Possibly even put the case classes into the same object if they all belong together.